python/car.py

```diff
@@ -32,3 +32,30 @@ def update_state(self, lane, distance_taken, speed, acceleration_state):
         self.speed = speed
         self.acceleration_state = acceleration_state
 
+    def get_distance_taken(self):
+        return self.distance_taken
+    
+    def get_follow_distance(self, safety_treshold=0.0):
+        # follow_distance = az a táv ami alatt meg tud állni 0-ra az autó az aktuális sebességről
+        # time to stop = speed / deceleration
+        # distance traveled = aree under the function of speed(time)
+        # which is a line from current speed at zero time, and zero speed at time to stop
+        follow_distance = (self.speed / 2.0) * (self.speed / self.specs.braking_power)
+        return (1 + safety_treshold) * follow_distance
+
+    #HELP
+    # distance taken while reaching target_speed from current_speed
+    def distance_while_accelerating(current_speed, target_speed, acceleration, deceleration):
+        _acceleration = 0
+        if current_speed < target_speed:
+            _acceleration = acceleration
+        else:
+            _acceleration = deceleration
```

Comment:
@torta211 Az if ág azért kell mert ugye más a car.specs-ben a gyorsulás meg a lassulás, viszont azért érdemes egy _acceleration változóban használni a relevánsat, mert fizikában ugyanaz a képlet negatív gyorsulásra is tudtommal, de lehet tévedek.

Reply:
Beletettem a kalkulációt, remélem helyes. Azt néztem, hogy itt igazából ha tagfüggvényként írjuk akkor nem kell beadni azokban a paramétereknek a nagy részét :D illetve a get függvény szerintem szükségtelen, pythonban ezek publikus adattagok tehát simán .distance_taken-nel le lehet kérdezni.

Reply:
illetve szerintem a safety-s paraméternél logikusabb ha maga a szorzótényező megy be. "biztonsági tényezőként" szokás hívni magyarul is